pkg/user/tx_client.go

```diff
@@ -445,26 +440,25 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 
 		if err == nil && resp != nil {
 			switch resp.Status {
-			// FIXME: replace hardcoded status with constants
-			case "PENDING":
+			case core.TxStatusPending:
 				// Continue polling if the transaction is still pending
 				select {
 				case <-ctx.Done():
 					return &TxResponse{}, ctx.Err()
 				case <-pollTicker.C:
 					continue
 				}
-			case "COMMITTED":
+			case core.TxStatusCommitted:
 				txResponse := &TxResponse{
 					Height: resp.Height,
 					TxHash: txHash,
 					Code:   resp.ExecutionCode,
 				}
 				if resp.ExecutionCode != 0 {
-					return txResponse, fmt.Errorf("tx was included but failed with code %d: %s", resp.ExecutionCode, resp.Status)
+					return txResponse, fmt.Errorf("tx was committed but failed with code %d: %s", resp.ExecutionCode, resp.Error)
 				}
 				return txResponse, nil
-			case "EVICTED":
+			case core.TxStatusEvicted:
 				return &TxResponse{}, fmt.Errorf("tx: %s was evicted from the mempool", txHash)
```

Comment:
sorry looks like I missed this in the first run.
```suggestion
				return &TxResponse{TxHash: txHash}, fmt.Errorf("tx: %s was evicted from the mempool", txHash)
```